Makefile

```diff
@@ -13,12 +13,12 @@ all: build test
 # Build target
 .PHONY: build
 build:
-	forge build
+	forge build --via-ir
```

Comment:
Hmm where is this coming from?  Are we exceeding compiler's threshold for happiness if we don't do this?  Can we instead remove local variables from functions?

Reply:
@ZenGround0 Very tricky to get rid of this . We need the local variables in `provePossession` to calculate gas fee and refactoring the code to get rid of these makes the code hard to read.

But I looked around and it's not necessarily bad  to use this flag i.e. it's not a smell/bad practise and is common with non-trivial contracts.

Reply:
Actually fixing this.

Reply:
Really slows down compilation.